homeassistant/components/demo/notify.py

```diff
@@ -1,27 +1,53 @@
 """Demo notification service."""
-from homeassistant.components.notify import BaseNotificationService
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.notify import BaseNotificationService, NotifyService
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.service_platform import AddServicesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
 EVENT_NOTIFY = "notify"
 
 
-def get_service(hass, config, discovery_info=None):
-    """Get the demo notification service."""
-    return DemoNotificationService(hass)
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_services: AddServicesCallback,
+) -> None:
+    """Set up the Demo config entry."""
+    async_add_services([DemoNotifyService()])
 
 
-class DemoNotificationService(BaseNotificationService):
-    """Implement demo notification service."""
+def get_service(
+    hass: HomeAssistant,
+    config: ConfigType,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> BaseNotificationService:
+    """Get the legacy demo notification service."""
+    return DemoNotificationService()
+
+
+class DemoNotifyService(NotifyService):
+    """Represent a demo notification service."""
 
-    def __init__(self, hass):
-        """Initialize the service."""
-        self.hass = hass
+    def send_message(self, message: str, **kwargs: Any) -> None:
+        """Send a message."""
+        kwargs["message"] = message
+        self.hass.bus.fire(EVENT_NOTIFY, kwargs)
+
+
+class DemoNotificationService(BaseNotificationService):
+    """Represent the legacy demo notification service."""
 
     @property
-    def targets(self):
+    def targets(self) -> dict[str, str]:
```

Comment:
One thing I notice here is that `targets` is no longer there. That is not ideal maybe?